src/app/utils/query-parameter-sanitization.ts

```diff
@@ -0,0 +1,415 @@
+// OData filter operators
+const QUERY_FUNCTIONS = ['startswith', 'endswith', 'contains', 'substring', 'indexof', 'concat'];
+const ARITHMETIC_OPERATORS = ['add', 'sub', 'mul', 'div', 'divby', 'mod'];
+const COMPARISON_OPERATORS = ['eq', 'ne', 'gt', 'ge', 'lt', 'le'];
+const LOGICAL_OPERATORS = ['and', 'or', 'not'];
+
+// REGEXES
+const ALL_ALPHA_REGEX = /^[a-z]+$/i;
+const POSITIVE_INTEGER_REGEX = /^[1-9]\d*$/;
+// Matches media type formats
+// Examples: https://www.iana.org/assignments/media-types/media-types.xhtml
+const MEDIA_TYPE_REGEX = /^(([a-z]+\/)?\w[\w+-.]+)$/i;
```

Comment:
I would like to propose the following instead, because having a `()` matches parts like `application/` as a separate group and we don't need that, So I removed the parentheses.

Also regarding words after the `/`, we can have "0 or more words" in addition to the first word, instead of "1 or more words". So I changed the last `+` to `*`.

Those changes break matching single word, which you previously had, so I added that with `|[a-z]+` to be more explicit about it.

Depending on how you consume this regex, you may need the outermost parentheses. Please make sure to run the unit tests :)

Please let me know if this makes sense.

```suggestion
const MEDIA_TYPE_REGEX = /^[a-z]+\/\w[\w+-.]*|[a-z]+$/i;
```

Reply:
Here are patterns that the original regex was supposed to match:
- `application/json`
- `application/1d-interleaved-parityfec`
- `application/atom+xml`

or their abbreviations
- `json`
- `1d-interleaved-parityfec`
- `atom+xml` 
(not sure if the last two abbreviations are true, but you get where I am going with this)

The outermost parentheses are required..

I get having the part `\w[\w+-.]+` like this `\w[\w+-.]*`  but not the first part about removing the parentheses. Could you please give more info if you can?

Reply:
ok, this makes sense. I thought you were matching application, not JSON. Thanks for the examples.